pkg/es/config/config.go

```diff
@@ -585,3 +608,198 @@ func (c *Configuration) Validate() error {
 	_, err := govalidator.ValidateStruct(c)
 	return err
 }
+
+// Run starts the cleaner that runs periodically based on the Frequency field.
+func (cfg *Configuration) RunCleaner() {
+	if !cfg.Cleaner.Enabled {
+		log.Println("Cleaner is disabled, skipping...")
+		return
+	}
+
+	ticker := time.NewTicker(cfg.Cleaner.Frequency)
+	defer ticker.Stop()
+
+	// Use for range to iterate over ticker.C channel
+	for range ticker.C {
+		log.Println("Running index cleaner...")
+		err := cfg.cleanIndices()
+		if err != nil {
+			log.Printf("Error cleaning indices: %v", err)
+		}
+	}
+}
+
+// cleanIndices connects to ElasticSearch and deletes the indices based on the configuration.
+func (cfg *Configuration) cleanIndices() error {
+	log.Println("Cleaning Elasticsearch indices...")
+
+	// Delete old indices (older than max span age)
+	err := cfg.deleteOldIndices()
+	if err != nil {
+		return fmt.Errorf("failed to delete old indices: %w", err)
+	}
+
+	// If IncludeArchive is true, also delete archived indices
+	if cfg.Cleaner.IncludeArchive {
+		err = cfg.deleteArchivedIndices()
+		if err != nil {
+			return fmt.Errorf("failed to delete archived indices: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// deleteOldIndices deletes indices older than the configured MaxSpanAge.
+func (cfg *Configuration) deleteOldIndices() error {
```

Comment:
doesn't this logic already exist in in cmd/es-index-cleaner? Why do we need to duplicate it?